apps/aecontract/src/aect_channel_contract.erl

```diff
@@ -76,7 +76,12 @@ run_new(ContractPubKey, Call, CallData, Trees0, OnChainTrees,
 prepare_init_call(VmVersion, Protocol, Contract, Trees0) when ?IS_FATE_SOPHIA(VmVersion) ->
     Code = #{ byte_code := ByteCode } = aeser_contract_code:deserialize(aect_contracts:code(Contract)),
     FateCode  = aeb_fate_code:deserialize(ByteCode),
-    FateCode1 = aeb_fate_code:strip_init_function(FateCode),
+    FateCode1 =
+        if
+        % Before FATE 2 we do not include init function in code put in trees
+            VmVersion == ?VM_FATE_SOPHIA_1 -> aeb_fate_code:strip_init_function(FateCode);
```

Comment:
BTW. It's a good question whether we shall enable the init function for FATE2 or after the IRIS hard fork.

Reply:
If we disable creation of FATE1 contracts after IRIS this will be enough.

Reply:
```
is_legal_version_at_protocol_(create, #{vm := ?VM_FATE_SOPHIA_1, abi := ?ABI_FATE_SOPHIA_1}, Protocol) ->
    case Protocol of
        ?ROMA_PROTOCOL_VSN            -> false;
        ?MINERVA_PROTOCOL_VSN         -> false;
        ?FORTUNA_PROTOCOL_VSN         -> false;
        ?LIMA_PROTOCOL_VSN            -> true;
        P when P > ?LIMA_PROTOCOL_VSN -> false
    end;
    ```

Reply:
We currently dissalow creation of FATE1 contracts on IRIS. I think there is no reason for anybody to deploy FATE1 after IRIS. The behaviour changes _should not_ affect anybody